src/blocks/CodeFeature/index.tsx

```diff
@@ -1,78 +1,145 @@
+  const tabWrapperRef = useRef<HTMLDivElement>(null)
+  const activeTabRef = useRef<HTMLButtonElement>(null)
+  const { heading, richText, forceDarkBackground, codeTabs, links } = codeFeatureFields
+  const id = useId()
 
-  let Spacer: React.ComponentType | 'div' = React.Fragment
+  useEffect(() => {
+    if (activeTabRef.current) {
+      setIndicatorStyle({
+        width: `${activeTabRef.current.clientWidth}px`,
+        left: `${activeTabRef.current.offsetLeft}px`,
+      })
+    }
+  }, [activeIndex])
 
-  const spacerProps: { className?: string } = {}
-
-  if (!disableBlockSpacing) {
-    Spacer = 'div'
-    spacerProps.className = classes.blockSpacing
-  }
+  useEffect(() => {
+    // Scroll logic has to sit in a separate useEffect because the wrapperWidth blocks the smooth scroll
+    if (activeTabRef.current) {
+      tabWrapperRef.current?.scroll(activeTabRef.current.offsetLeft - 20, 0)
+    }
+  }, [activeIndex])
 
   return (
-    <div className={classes.codeFeature}>
-      <Spacer {...spacerProps}>
-        <Gutter>
-          <div className={[classes.codeFeatureGrid, 'grid'].filter(Boolean).join(' ')}>
-            <div className={'cols-8 cols-m-8'}>
-              <h2 className={classes.heading}>{heading}</h2>
-              <div className={[classes.innerCodeFeatureGrid, 'grid'].filter(Boolean).join(' ')}>
-                <div className={`cols-6 start-${disableIndent ? 1 : 2} cols-m-8 start-m-1`}>
-                  <RichText content={richText} className={classes.richText} />
-                  {enableLink && <CMSLink {...link} />}
+    <div
+      className={[classes.wrapper, forceDarkBackground && classes.darkTheme]
+        .filter(Boolean)
+        .join(' ')}
+      {...(forceDarkBackground ? { 'data-theme': 'dark' } : {})}
+      id={id}
+    >
+      <Gutter>
+        <div className={[classes.container, 'grid'].filter(Boolean).join(' ')}>
+          <BackgroundGrid ignoreGutter className={classes.backgroundGrid} />
+          <div className={[classes.scanlineWrapper, 'start-9 cols-8'].filter(Boolean).join(' ')}>
+            <BackgroundScanline
+              className={[classes.scanlineDesktop].filter(Boolean).join(' ')}
+              crosshairs={['top-left', 'bottom-left']}
+            />
+            {/* <CrosshairIcon className={[classes.crosshairTopLeft].filter(Boolean).join(' ')} /> */}
+            <CrosshairIcon className={[classes.crosshairTopRight].filter(Boolean).join(' ')} />
+            {/* <CrosshairIcon className={[classes.crosshairBottomLeft].filter(Boolean).join(' ')} /> */}
+            <CrosshairIcon className={[classes.crosshairBottomRight].filter(Boolean).join(' ')} />
+          </div>
+          <div className={[classes.content, 'cols-4 cols-m-8'].filter(Boolean).join(' ')}>
+            <h2 className={classes.heading}>{heading}</h2>
+            {forceDarkBackground}
+            <div className={[''].filter(Boolean).join(' ')}>
+              <div className={''}>
```

Comment:
@paulpopus remove `className` from these if there are no classes being added to the elements & are these wrapping elements even needed then?